Reporting/M5 Document/StepByStep.md

```diff
@@ -0,0 +1,97 @@
+
+        -   Accept the invite to my project on Atlas
+
+        -   In the Frontend/ directory, create a .env file and follow
+            > the instructions listed at the bottom of this doc
+
+        -   Test your MongoDB connection to the frontend by having the
+            > development server for Next.js up (npm run dev) and then
+            > navigating to http://localhost:3000/api/testMongoDB. You
+            > should see the following if your connection is successful:
+            > {\"message\":\"Successfully connected to MongoDB!\"}
+
+        -   Test your MongoDB connection to the backend by executing
+            > python manage.py under the backend directory. At the top
+            > of the list of available commands, you should see the
+            > following if your connection is successful: Pinged your
+            > deployment. You successfully connected to MongoDB!. Note:
+            > I configured Django so that the database connection is
+            > automatically established every time any Python file is
+            > executed.
+
+        -   .env file setup
+
+            -   Firstly, make sure you have the following installed:
+                > mongodb and dotenv. If not, do npm install mongodb and
+                > npm install dotenv or use yarn.
+
+            -   Inside the .env file, place the following content:
+
+                -   **MONGODB_USERNAME=\<username\>**
+
+                -   **MONGODB_PASSWORD=\<password\>**
+
+                -   **MONGODB_URI=mongodb+srv://\<username\>:\<password\>@accord-systems.umbugbv.mongodb.net/?retryWrites=true&w=majority&appName=Accord-Systems**
+
+            -   Set MONGODB_USERNAME and MONGODB_PASSWORD to the
+                > credentials for your Atlas account that I gave to you.
+                > Do not change replace \<username\> and \<password\> in
+                > MONGODB_URI. That will be replaced with a script that
+                > runs when the database connection is established in
+                > Next.js. For windows users, open PowerShell as
+                > administrator and execute New-Item -ItemType
+                > SymbolicLink -Path .\\.env -Target ..\\.env. This is
+                > needed because Django can access that .env file but
+                > Next.js can\'t as it is outside of the Next.js app
+                > (Frontend directory).
+
+            -   NOTE: do not include the angle brackets (\<, \>).
+
+            -   You do not need to worry about accidentally committing
+                > your .env file as it is already listed in .gitignore.
+
+    -   Ably
+
+        -   Please add the following to your .env file
+
+            -   **ABLY_API_KEY_PUBLISH_SUBSCRIBE=yourapikey**
+
+            -   Replace yourapikey with the key given to you.
+
+        -   You will need to run this command before you start up the
+            > application: **ably: npm install ably**
```

Comment:
We should have this as a dependency in `package.json` so we don't have to write this. All the user should have to do is `npm install`, but we need to make sure `package.json` has all our dependencies marked within it. Right now I don't think ably is included in there. The way we would solve this is by running `npm install -D myPackage` for every package that we are using (the core ones like ably and mongodb) so that we are sure they are marked as dependencies.